pkg/prometheus/promcfg.go

```diff
@@ -585,13 +585,16 @@ func (cg *ConfigGenerator) addProxyConfigtoYaml(
 	}
 
 	if proxyConfig.ProxyConnectHeader != nil {
-		proxyConnectHeader := make(map[string]string, len(proxyConfig.ProxyConnectHeader))
+		proxyConnectHeader := make(map[string][]string, len(proxyConfig.ProxyConnectHeader))
 
 		for k, v := range proxyConfig.ProxyConnectHeader {
-			value, _ := store.GetKey(ctx, namespace, monitoringv1.SecretOrConfigMap{
-				Secret: &v,
-			})
-			proxyConnectHeader[k] = value
+			proxyConnectHeader[k] = []string{}
+			for _, s := range v {
+				value, _ := store.GetKey(ctx, namespace, monitoringv1.SecretOrConfigMap{
```

Comment:
I'm not sure why we use `monitoringv1.SecretOrConfigMap{}` here. Can we change to store.GetSecretKey?

Reply:
Thanks. Done.